### ISSUE TITLE
Handling of Dry and Partially Saturated Cells under NR solution

### ISSUE THREAD

**Issue by askar-INTERA** [MODFLOW-USGS/modpath-v7#36]
I wonder how the latest version of Modpath-7 (the provisional version) handles the dry and partially saturated cells in a MF6 Newton Raphson solution. 

The two attached figs (the dashed lines = cell bottom elevation, solid lines = cell head, and colors = different units/formations) represent two different cases/pathways of two particles moving from a contamination source at the surface (tailings) downward through some dry and partially saturated units. 
As you can see in the upper fig, the particle dropped from the bottom of the tailings to the bottom of alluvium almost instantly because both were dry, I guess, correct? if yes, is that what we expect to physically happen in the site? 
However, in the lower figure you can see that the particle was dropped from the bottom of alluvium to the water table in TRA almost instantly, meaning that the particle is not really affected by the dry part of the cell, correct?

For the upper case, I tried to calculate the time the particle should take to go through the cell thickness (reaching the bottom elevation of the alluvium) using the recorded flows in the cbb file with the equations posted in Modpath-7 docs, but I was not able to get the same value calculated by Modpath-7 (Maybe I am doing some wrong, not sure).

In summary my question is: how travel time was calculated for these cases? Using which velocity/flow components? 

![case_1](https://github.com/MODFLOW-USGS/modpath-v7/assets/96551098/6d1b9b5c-3579-4481-a5e9-74ff7382cf98)
![case_2](https://github.com/MODFLOW-USGS/modpath-v7/assets/96551098/c86b52dc-6558-4a57-8395-5606b037abcd)

**Comment by aprovost-usgs** [comment]
Hi @askar-INTERA. I saw your related post in issue #2 and had an initial look into the code.  My preliminary conclusion was that MODPATH 7 is not equipped to handle the case of a particle in a Newton dry cell, at least not properly or gracefully, but I wanted to look into it further before responding.  Thanks very much for the additional information - it should be very helpful, though I might have some addition questions if you don't mind.  I'll look into it further and post an assessment as soon as possible.

**Comment by askar-INTERA** [comment]
Thank you so much @aprovost-usgs. We can also arrange a meeting to show you more about our model and how the particles are behaving under NR solution and also our hand calculation for the travel time.

**Comment by aprovost-usgs** [comment]
@askar-INTERA Just wanted to touch base and let you know that we haven't forgotten about this issue.  We're also looking into a related issue of particles moving through "perched" cells.

**Comment by askar-INTERA** [comment]
@aprovost-usgs thanks for your effort, I really appreciate it. It would be a great win if we can find a way so that Modpath can deal with the flow field generated by NR method and somehow introduce some decay to the velocity of the particles going through dry (always active) cells and also through the dry part of partially saturated cells (or cells simulating perched conditions).

**Comment by askar-INTERA** [comment]
@aprovost-usgs Hello, just checking if you guys came to any solution to the above issue, even if it is temp solution. Thank you so much.

**Comment by aprovost-usgs** [comment]
@askar-INTERA Hello, and thanks for your patience. According to a colleague who recently looked into a similar issue with a different model, the travel times through dry Newton cells were not instantaneous and were consistent with the flows being passed vertically through those cells. However, I've taken a close look (using the trace option) at your particle with sequence number 3258 as it passes through dry cell 14003, and while the travel is not instantaneous, the rates (flows, velocities, times) don't seem to be adding up. I'll continue looking into it today and tomorrow, and I'd be interested to know how you approached your travel time estimates if you wouldn't might describing that to me.

**Comment by askar-INTERA** [comment]
@aprovost-usgs the PT records show that several mins are needed to go through the tailings thickness (if I remember correctly), but when I pulled out the volumetric rate across the lower face of that cell from the cbb file (which basically equals the upper face volumetric rate and both equal the applied recharge, as the cell is dry) and then used the below equation (from the MP docs) to calculate the velocity and finally divided the cell thickness by that velocity to calculate the travel time, I got drastically different answer compared to MP results. Am I doing something wrong?
But anyway, the particle shouldn't take several mins only to go through a thick unit with a permeability of 1.6 ft/d, even if the cell is dry and thus Newton Raphson is giving a non-realistic downward flux, correct?
![image](https://github.com/MODFLOW-USGS/modpath-v7/assets/96551098/f756a196-32d6-4dce-9c09-1f37b6adf7cb)

**Comment by aprovost-usgs** [comment]
@askar-INTERA  Thanks for your reply.  Your velocity formula is the same as what I'm using, and the travel times do seem much too fast.  What appears to be going on is that in a dry but active cell MODPATH sets the fractional saturation to an arbitrary, small value of 1d-4.  The idea appears to be to move particles through dry but active cells (almost) instantaneously using a simple approach that doesn't require a lot of specialized coding.  In other words, to use the existing tracking algorithm but adjust the distance the particle has to travel so it's really short.  In that case the exact travel time doesn't have much meaning, except that it's "close to zero."  We'll take another dive into the code to satisfy ourselves that this is how it was intended to work, and that there aren't any other issues.  Then we'll need to decide how to proceed.  Near-instantaneous propagation through dry cells might be ok for some applications, but for others it would be good if the travel times were representative of the finite flow rates through those cells, even if those flow rates are themselves approximate.  We'll think it through and see if we can come up with a reasonable way to implement this in MODPATH.

**Comment by askar-INTERA** [comment]
@aprovost-usgs @wpbonelli I am just curious to know whether MF6 PRT handles dry cells differently than ModPath7 under NR solution?

**Comment by aprovost-usgs** [comment]
@askar-INTERA PRT, which is still under development, currently passes particles through dry but active cells instantaneously.  As I mentioned earlier, it appears that's what MODPATH 7 is effectively doing in an approximate way, by setting a very small cell saturation.  PRT passes particles through dry but active cells directly, without setting an artificial value for cell saturation.  That said, we're considering whether there's a reasonable way to do better and make the particle travel times more representative of the finite flow rates passing through dry but active cells in both MODPATH 7 and PRT.
